WooCommerce/Classes/Tools/Currency/CurrencySettings.swift

```diff
@@ -132,6 +130,7 @@ public class CurrencySettings {
                   thousandSeparator: CurrencySettings.Default.thousandSeparator,
                   decimalSeparator: CurrencySettings.Default.decimalSeparator,
                   numberOfDecimals: CurrencySettings.Default.decimalPosition)
+        configureResultsController()
```

Comment:
> In CurrencySettings init with parameters for all the properties, removed the side effect of configureResultsController() and called configureResultsController() in the convenient init() that is only used in the app shared instance

I get the feeling that this `ResultsController` refreshing should be somewhere else instead. But I'm not sure how we can do that. 🤔 

A slight concern with this change is that the class is inconsistently initialized depending on what `init` you use. One way we could probably avoid this is to make the `lazy var resultsController` smarter. 

1. Change `configureResultsController` to:

    ```swift
        private func configure(resultsController: ResultsController<StorageSiteSetting>) {
    ```
2. Change `refreshResultsPredicate` to: 

    ```swift
        private func refreshPredicate(resultsController: ResultsController<StorageSiteSetting>) {
    ```
3. Change the `lazy var` so it is truly _lazily configured_:

    ```swift
        private lazy var resultsController: ResultsController<StorageSiteSetting> = {
            let storageManager = ServiceLocator.storageManager
            let descriptor = NSSortDescriptor(keyPath: \StorageSiteSetting.siteID, ascending: false)
            let resultsController = ResultsController<StorageSiteSetting>(storageManager: storageManager, sortedBy: [descriptor])

            configure(resultsController: resultsController)

            return resultsController
        }()
    ```
4. Remove the `configureResultsController` call from the `init()`. 

So now the two `init` methods result in the same behavior. 

This is just a thought exercise as we'd have to be careful with a change such as this since `CurrencySettings` is a singleton. 🙂 So, I wouldn't recommend a change like this in this PR.

Reply:
my top choice would be 4. `Remove the configureResultsController call from the init().` - I felt like the call to configure the results controller is a side effect (overriding the input parameters) that isn't expected. though, we usually don't call a configure method from outside of a struct/class 🤔maybe we can open another function like `startListeningToSiteSettings` that calls `configureResultsController`?

Reply:
> maybe we can open another function like startListeningToSiteSettings that calls configureResultsController?

With this, `CurrencySettings` is dumb by default and is only "active" when we call `startListeningToSiteSettings`. That sounds better to me!